src/network/deployer/manifest.ts

```diff
@@ -0,0 +1,246 @@
+        return (await this.read()).admin;
+    }
+
+    async getDeploymentFromAddress(address: string): Promise<ImplDeployment> {
+        const data = await this.read();
+        const deployment = Object.values(data.impls).find(
+            d => d?.address === address || d?.allAddresses?.includes(address)
+        );
+
+        if (deployment === undefined) {
+            throw new DeploymentNotFound(`Deployment at address ${address} is not registered`);
+        }
+        return deployment;
+    }
+
+    async getProxyFromAddress(address: string): Promise<ProxyDeployment> {
+        const data = await this.read();
+        const deployment = data.proxies.find(d => d?.address === address);
+        if (deployment === undefined) {
+            throw new DeploymentNotFound(`Proxy at address ${address} is not registered`);
+        }
+        return deployment;
+    }
+
+    async addProxy(proxy: ProxyDeployment): Promise<void> {
+        await this.lockedRun(async () => {
+            const data = await this.read();
+            const existing = data.proxies.findIndex(p => p.address === proxy.address);
+            if (existing >= 0) {
+                data.proxies.splice(existing, 1);
+            }
+            data.proxies.push(proxy);
+            await this.write(data);
+        });
+    }
+
+    private async exists(file: string): Promise<boolean> {
+        try {
+            await fs.access(file);
+            return true;
+        } catch (e: any) {
+            return false;
+        }
+    }
+
+    private async readFile(): Promise<string> {
+        if (this.file === this.fallbackFile) {
+            return await fs.readFile(this.file, 'utf8');
+        } else {
+            const fallbackExists = await this.exists(this.fallbackFile);
+            const fileExists = await this.exists(this.file);
+
+            if (fileExists && fallbackExists) {
+                throw new UpgradesError(
+                    `Network files with different names ${this.fallbackFile} and ${this.file} were found for the same network.`,
+                    () =>
+                        `More than one network file was found for chain ID ${this.chainId}. Determine which file is the most up to date version, then take a backup of and delete the other file.`
+                );
+            } else if (fallbackExists) {
+                return await fs.readFile(this.fallbackFile, 'utf8');
+            } else {
+                return await fs.readFile(this.file, 'utf8');
+            }
+        }
+    }
+
+    private async writeFile(content: string): Promise<void> {
+        await this.renameFileIfRequired();
+        await fs.writeFile(this.file, content);
+    }
+
+    private async renameFileIfRequired() {
+        if (this.file !== this.fallbackFile && (await this.exists(this.fallbackFile))) {
+            try {
+                await fs.rename(this.fallbackFile, this.file);
+            } catch (e: any) {
```

Comment:
If you expect `e.message` one line below, then the type should be `Error`. The same applies to line [112](https://github.com/pointnetwork/pointnetwork/pull/700/files#diff-d36c130e91bc0f069dc555635e57a89704a1133e9c2dc3f3cd36dd812fbd346bR112) and probably [158](https://github.com/pointnetwork/pointnetwork/pull/700/files#diff-d36c130e91bc0f069dc555635e57a89704a1133e9c2dc3f3cd36dd812fbd346bR158).